src/node.cc

```diff
@@ -5337,23 +5337,28 @@ v8::Local<v8::Object> IncludeModule(const std::string& module_name) {
   return v8::Local<v8::Object>::Cast(module);
 }
 
-void RegisterModule(const std::string & name, const addon_context_register_func & callback, void *priv) {
-    node::node_module* module = new node::node_module();
+void RegisterModule(const std::string & name, const addon_context_register_func & callback, void *priv, const bool auto_bind) {
+  node::node_module* module = new node::node_module();
 
-    module->nm_version = NODE_MODULE_VERSION;
-    module->nm_flags = NM_F_BUILTIN;
-    module->nm_filename = __FILE__;
-    module->nm_context_register_func = callback;
-    module->nm_modname = name.c_str();
-    module->nm_priv = priv;
+  module->nm_version = NODE_MODULE_VERSION;
+  module->nm_flags = NM_F_BUILTIN;
+  module->nm_filename = __FILE__;
+  module->nm_context_register_func = callback;
+  module->nm_modname = name.c_str();
+  module->nm_priv = priv;
 
-    node_module_register(module);
+  node_module_register(module);
+
+  if(auto_bind) {
+    Evaluate(name + " = process.binding('" + name + "')");
```

Comment:
Hmm. Should we check `name` for invalid characters?

Reply:
from what I can see, nodejs' `Binding` function (defined in node.cc, invoked by process.binding()) throws a JS exception when no module named `name` is found, so this check is made there already.

ATM, I don't really know how node's exceptions work, so if you don't have a solution to handling these at this moment, I'd say we attach it to issue #17 for now.

Reply:
or open a new issue for exception handling (vs. MaybeLocal stuff)

Reply:
I was thinking about the case when invalid js code is executed:
```cpp
std::name name = "123-foo"; // or "while(true) {}; foo"
Evaluate(name + " = process.binding('" + name + "')");
// Evaluate("123-foo = process.binding('123-foo')");
// Error: 123-foo isn't allowed as variable name.
```

Also, maybe add `const` in front of the variable name in JS.

Reply:
Tried it out, throws a [SyntaxError](https://nodejs.org/api/errors.html#errors_class_syntaxerror) from node.js. Which makes sense, since all errors produced by javascript code (even by faulty JS code), result in JS exceptions/errors.

So again, (IMO) this should still be handled with these exceptions just like with every other call to `Evaluate()`.

`const` is a good call.